pkg/driver/driver_azure.go

```diff
@@ -163,6 +163,15 @@ func (d *AzureDriver) getVMParameters(vmName string, networkInterfaceReferenceID
 		}
 	}
 
+	if d.AzureMachineClass.Spec.Properties.IdentityID != nil {
```

Comment:
` && *d.AzureMachineClass.Spec.Properties.IdentityID != ""`?

Reply:
Thanks I have adapted this.